tests/std/tests/P0083R3_splicing_maps_and_sets/test.cpp

```diff
@@ -634,6 +658,143 @@ void test_unordered_set() {
     test_merge<Set, std::unordered_multiset>();
 }
 
+void test_gh1309() {
+    // Guard against regression of GH-1309, in which node handles were incorrectly destroying the user value with a node
+    // allocator rather than a value_type allocator as the Standard requires.
+
+    {
+        using A  = tracked_allocator<std::pair<int const, char>>;
+        using M  = std::map<int, char, std::less<>, A>;
+        using NH = decltype(std::declval<M&>().extract(42));
+        allocation_guard guard{true};

```

Comment:
@mnatsuhara noticed that the `allocation_guard` machinery used in the previous test code, isn't performing any useful work here (as there are no regions where we want to prohibit allocation/deallocation - because the bug is unrelated). I think it would be clearer to construct a single guard for the entire body of this test function.